pyfar/dsp/dsp.py

```diff
@@ -1498,3 +1498,144 @@ def decibel(signal, domain='freq', log_prefix=None, log_reference=1,
         return log_prefix * np.log10(np.abs(data) / log_reference), log_prefix
     else:
         return log_prefix * np.log10(np.abs(data) / log_reference)
+
+
+def energy(signal, keepdims=False):
+    r"""
+    Compute the energy of a signal.
+
+    According to the Parseval's theorem [#]_ the energy is
+
+    .. math::
+
+        \sum_{n=0}^{N-1}|x[n]|^2=\frac{1}{N}\sum_{k=0}^{N-1}|X[k]|^2,
+
+    The function calculates the energy from the time data for each channel
+    separately.
+
+    Parameters
+    ----------
+    signal : Signal
+        The signal to compute the energy from.
+    keepdims : bool, optional
+        If ``False``, the shape of the returned array equals ``signal.cshape``.
+        If ``True``, an axis will be appended, so it's broadcastable to the
+        shape of ``signal.time`` and ``signal.freq``.
+        The default is ``False``.
+
+    Returns
+    -------
+    data : numpy.ndarray
+        The channel-wise energy of the input signal.
+
+    Notes
+    -----
+    Due to the calculation based on the time data, the returned energy is
+    independent of the signal's ``fft_norm``.
+    :py:func:`~pyfar.dsp.power` and :py:func:`~pyfar.dsp.rms` can be used
+    to compute the power and the rms of a signal.
+
+    References
+    -----------
+    .. [#] B. Delgutte and J. Greenberg , “The discrete Fourier
+           Transform”, Biomedical Signal and Image Processing, Spring 2005.
```

Comment:
I'd rather use literature concerned with audio signal processing, as people will have that more readily available.